frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -561,9 +576,69 @@ function ReaderHighlight:_resetHoldTimer(clear)
     end
 end
 
+function ReaderHighlight:getRegionalZoomCenter(pos)
+    if not self.dimen then
+        self.dimen = self.ui.dimen
+    end
+    local page_size = self.ui.document:getNativePageDimensions(pos.page)
+    local margin = self.ui.document.configurable.page_margin * Screen:getDPI()
+    local zoom = 2*self.dimen.w / page_size.w
+    zoom = zoom/(1 + 3*margin/zoom/page_size.w)
+    local x_ratio = (pos.x/ page_size.w) -- x center ratio
+    local y_ratio = (pos.y / page_size.h) -- y center ratio
+       
+    return zoom, x_ratio, y_ratio
+    
```

Comment:
These kind of blank lines around the return are not really - they just make the reader scroll more :)
(I may use one to separate different logical works in a same function.)

Reply:
done.
I need to change my habit :)